tests/drivers/i2s/i2s_api/src/main.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2017 comsuisse AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @addtogroup t_driver_i2s
+ * @{
+ * @defgroup t_i2s_basic test_i2s_basic_operations
+ * @}
+ *
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+
+void test_i2s_tx_transfer_configure_0(void);
+void test_i2s_rx_transfer_configure_0(void);
+void test_i2s_transfer_short(void);
+void test_i2s_transfer_long(void);
+void test_i2s_rx_empty_timeout(void);
+void test_i2s_transfer_restart(void);
+void test_i2s_transfer_rx_overrun(void);
+void test_i2s_transfer_tx_underrun(void);
+
+void test_i2s_tx_transfer_configure_1(void);
+void test_i2s_rx_transfer_configure_1(void);
+void test_i2s_state_not_ready_neg(void);
+void test_i2s_state_ready_neg(void);
+void test_i2s_state_running_neg(void);
+void test_i2s_state_stopping_neg(void);
+void test_i2s_state_error_neg(void);
```

Comment:
I placed function declarations here following an example of other tests. However, that does not seem very handy. Maybe we should place all these function declarations in a separate header file? Or define `ztest_test_suite` in a testcase .c file? E.g. place `ztest_test_suite(i2s_loopback_test, ...);` in test_i2s_loopback.c so there is no need to write test function declarations.